main.cpp

```diff
@@ -121,196 +121,226 @@ struct CarWash
     float waterPressure = 5.0f;
-    // Size of container
     int containerSize = 2;
-    // volume of soda tank 
     float sodaTankVolume = 0.5f;
-    // years of guarantee
-    int guaranteeYears = 2;
-    // number of bottles
-    int numBottles = 1;
-
-    // make soda water
-    void makeSodaWater(int pressButtonDuration);
-    // install soda tank
+    int sparklingWaterFactor = 2;
+    bool sodaTankEmpty = false;
+
+    void makeSodaWater(float waterVolume);
     void installSodaTank();
-    // get name of the manufacturer
     std::string getManufacturerName();
 };
 
 
+void SodaMaker::makeSodaWater(float waterVolume)
+{
+    if(!sodaTankEmpty && waterVolume <= sodaTankVolume)
+    {
+        // Create wonderful sparkling water!
+    }
+}
+
+
+void SodaMaker::installSodaTank()
+{
+    sodaTankEmpty = false;
+}
 
-/*
-Thing 3) Dish Washer
-5 properties:
-    1) volume of washing room
-    2) number of programs
-    3) duration of duration of the washing process
-    4) amount of detergen tank
-    5) duration of timer
-3 things it can do:
-    1) wash dishes
-    2) dry dishes
-    3) start at timer
- */
+
+std::string SodaMaker::getManufacturerName() {
+    return "SodaStream";
+}
+
+
+//--------------------------------------------------------
+// DishWasher
+//--------------------------------------------------------
 
  struct DishWasher
  {
-    // volume of washing room
     float washingRoomVolume = 5.2f;
-    // number of programs
     int numOfPrograms = 12;
-    // duration of duration of the washing process
-    int washingDuration = 120;
-    // amount of detergen tank
-    float amountDetergenTank = 0.200f;
-    // duration of timer
+    int maxWashingDuration = 120;
+    float maxWashTemperature = 90.0f;
     int timerDuration = 360;
+    int waitMilliSec = 0;
 
-    // wash dishes (returns true after the sucessful process)
     bool washDishes(int programNumber);
-    // dry dishes (returns true after the sucessful process)
-    bool dryDishes(int duration, int temperature);
-    // start at timer
+    bool dryDishes(int duration, float temperature);
     void startAtTimer(int startMinutesLater);   
  };
 
-/*
-Thing 4) Gas Grill
-5 properties:
-    1) size of grill plate
-    2) volume of coal
-    3) number level positions
-    4) heat temperature
-    5) port size of gas tank 
-3 things it can do:
-    1) grill sausage
-    2) light the flame
-    3) turn of the flame
- */
+
+bool DishWasher::washDishes(int programNumber)
+{
+    if(programNumber > 12) return false;
+    // Wash dishes!
+    return true;
+}
+
+
+bool DishWasher::dryDishes(int duration, float temperature)
+{
+    if(duration <= maxWashingDuration || temperature <= maxWashTemperature) 
+    {
+        // Drying!
+        return true;
+    }
+    return false;
+}
+
+
+void DishWasher::startAtTimer(int startMinutesLater)
+{
+    waitMilliSec = startMinutesLater * 1000;
+    // sleep(startMinutesLater * startMinutesLater);
+    DishWasher::washDishes(1);    
```

Comment:
There is no need for the fully-qualified name here
`washDishes` is not a static member function
- [ ] revise